libs/coin-evm/src/api/transaction/index.ts

```diff
@@ -0,0 +1,15 @@
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { getExplorerApi } from "../explorer";
+
+// TODO: implement getTransactionByHash for etherscan explorer
+export const getTransactionByHash = async (
+  currency: CryptoCurrency,
+  transactionHash: string,
+): Promise<{ confirmations?: number }> => {
+  const { getTransactionByHash: apiGetTransactionByHash } = getExplorerApi(currency);
+  if (!apiGetTransactionByHash) {
+    return {};
```

Comment:
Shouldn't this throw an error and then we use a catch where it's called with the corresponding error ?

Reply:
Yeah seems like `getExplorerApi` can throw an error, it should be wrapped around a try/catch: https://github.com/LedgerHQ/ledger-live/blob/feat/LIVE-8957-migrate-speedup-cancel-to-evm/libs/coin-evm/src/api/explorer/index.ts#L23

Reply:
This file has been rewritten, cf. https://github.com/LedgerHQ/ledger-live/pull/4886#discussion_r1352513038

cf. 9320e023af14319ab3ad4a6ef350426aa5f6d195